compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmDesugarPhase.java

```diff
@@ -160,275 +154,119 @@ private static void rewriteRecordInitFunction(BIRFunction func, BRecordType reco
+                parameter.name = encodeNonFunctionIdentifier(parameter.name.value);
             }
             if (function.type.tsymbol != null) {
                 for (BVarSymbol parameter : ((BInvokableTypeSymbol) function.type.tsymbol).params) {
-                    parameter.name = getInitialIdString(parameter.name, encodedVsInitialIds);
+                    parameter.name = encodeNonFunctionIdentifier(parameter.name.value);
                 }
             }
-            replaceEncodedDefaultFunctionName(function.type, encodedVsInitialIds);
-            replaceEncodedWorkerName(function, encodedVsInitialIds);
+            encodeDefaultFunctionName(function.type);
+            encodeWorkerName(function);
         }
     }
 
-    private static void replaceEncodedDefaultFunctionName(BInvokableType type,
-                                                          HashMap<String, String> encodedVsInitialIds) {
+    private static void encodeDefaultFunctionName(BInvokableType type) {
         BInvokableTypeSymbol typeSymbol = (BInvokableTypeSymbol) type.tsymbol;
         if (typeSymbol == null) {
             return;
         }
         for (BInvokableSymbol defaultFunc : typeSymbol.defaultValues.values()) {
-            defaultFunc.name = getInitialIdString(defaultFunc.name, encodedVsInitialIds);
+            defaultFunc.name = encodeFunctionIdentifier(defaultFunc.name.value);
         }
     }
 
-    private static void replaceEncodedWorkerName(BIRFunction function,
-                                                 HashMap<String, String> encodedVsInitialIds) {
+    private static void encodeWorkerName(BIRFunction function) {
         if (function.workerName != null) {
-            function.workerName = getInitialIdString(function.workerName, encodedVsInitialIds);
+            function.workerName = encodeNonFunctionIdentifier(function.workerName.value);
         }
     }
 
-    private static void replaceEncodedAttachedFunctionIdentifiers(List<BAttachedFunction> functions,
-                                                                  HashMap<String, String> encodedVsInitialIds) {
+    private static void encodeAttachedFunctionIdentifiers(List<BAttachedFunction> functions) {
         for (BAttachedFunction function : functions) {
-            function.funcName = getInitialIdString(function.funcName, encodedVsInitialIds);
+            function.funcName = encodeFunctionIdentifier(function.funcName.value);
         }
     }
 
-    private static void replaceEncodedGlobalVariableIdentifiers(List<BIRNode.BIRGlobalVariableDcl> globalVars,
-                                                                HashMap<String, String> encodedVsInitialIds) {
+    private static void encodeGlobalVariableIdentifiers(List<BIRNode.BIRGlobalVariableDcl> globalVars) {
         for (BIRNode.BIRGlobalVariableDcl globalVar : globalVars) {
+            // globalVar should not be null, check and remove this
```

Comment:
Are we going to address this in the same PR? If not it's better to create a separate task and mention it as TODO. I'll be easier to track.

Reply:
Removed this.